pytorch_lightning/plugins/environments/lightning_environment.py

```diff
@@ -34,6 +35,8 @@ class LightningEnvironment(ClusterEnvironment):
     def __init__(self):
         super().__init__()
         self._master_port = None
+        self._global_rank: int = 0
+        self._world_size: Optional[int] = None
```

Comment:
How can a world size be None? Isn't it 1 then? Also what happens if this is None? Do we check this?

Reply:
yes, we test it in tests/plugins/environments/test_lightning_environments
you are right, it could be 1 by default now.
we never see None because at trainer init we immediately let the plugin overwrite it to an int.